bcgov/NRPTI
# NRPT-697: added name redaction on issuingAgency and made datepicker years scrollable

[NRPT-697](https://bcmines.atlassian.net/browse/NRPT-697): this is a recreation of the last PR to cleanup the commits 

To test switch your role to admin:nrced or admin:lng only and create a record with issued to type individual, age over 19 and publish to NRCED or LNG. Verify that the created records have public in the base read, but not in the issued to read.

for the following issuing agencies you should see public in the issuedTo read array if the other age requirements are met:

`
exports.AUTHORIZED_PUBLISH_AGENCIES = [
  'BC Parks',
  'Climate Action Secretariat',
  'Conservation Officer Service (COS)',
  'EAO',
  'Environmental Protection Division'
];
`

Any edits should apply the correct roles to the issuedTo read array as well.

To test the migration, you can use:

`
db.nrpti.find({ issuingAgency: {$in: ["BC Oil and Gas Commission","Ministry of Energy, Mines and Low Carbon Innovation","Natural Resource Officers","Ministry of Forests, Lands, and Natural Resource Operations","Agricultural Land Commission","Ministry of Agriculture","BC Wildfire Service"]}, "issuedTo.type": "Individual","issuedTo.read": { $all: ["public"] }} ).pretty()
`

and then there should be none after you run the migration.

I also updated all of the datepickers to at least minDate vars. I checked with Kyle that dateIssued should not be able to be set in the future so added maxDate to those ones as well as the birthday datepickers.

Most of the datepickers are in admin nrpti, but there is one in public nrced
